### PR TITLE
feat: Replace Moya with Alamofire

### LINKED ISSUE #11: Replace Moya with Alamofire
Reduce the dependency graph by using Alamofire instead of Moya. 

```
Moya -> Almofire -> URLSession
Alamofire -> URLSession 
```

### PR DESCRIPTION
**Pull request by ahmdyasser**
I replaced Moya with Alamofire to reduce the dependency in the project.

### CODE REVIEW

**Review by ahmednaagy — APPROVED** [review]
Great I was just thinking about making these changes, Good job 💯

**Review by oradwanomar — APPROVED** [review]

**Review by oradwanomar — COMMENTED** [review]
Perfecto 👌🏻

File: Networking/Package.swift [hunk 1]
```diff
@@ -18,15 +18,15 @@ let package = Package(
     dependencies: [
         // Dependencies declare other packages that this package depends on.
         // .package(url: /* package url */, from: "1.0.0"),
-        .package(url: "https://github.com/Moya/Moya.git", .upToNextMajor(from: "15.0.0")),
+        .package(url: "https://github.com/Alamofire/Alamofire.git", .upToNextMajor(from: "5.6.1")),
```
Comment (ahmdmhasn) [inline Networking/Package.swift:21]: What's the difference between 
```swift
.package(url: "https://github.com/Alamofire/Alamofire.git", .upToNextMajor(from: "5.6.1")),
```
and 
```swift
.package(url: "https://github.com/Alamofire/Alamofire.git", .upToNextMajor(from: "5.0.0")),
```
@ahmdyasser 😁

Comment (ahmdyasser) [reply Networking/Package.swift:21]: @ahmdmhasn I just took the latest stable version form GitHub, did it cause any problem?

**Pull request merged**